cpp/Platform.Equality.Tests/Platform.Equality.Benchmark.cpp

```diff
@@ -0,0 +1,126 @@
+#include <Platform.Equality.h>
+#include <benchmark/benchmark.h>
+
+#include <ctime>
+#include <iostream>
+#include <span>
+#include <any>
+
+namespace Platform::Equality::Benchmark
+{
+    namespace Any
+    {
+        static void AnyCompare(benchmark::State& state)
+        {
+            srand(time(nullptr));
+
+            for (auto _ : state)
+            {
+                auto a = rand();
+                auto b = rand();
+
+                bool result;
+                benchmark::DoNotOptimize(result = std::any(a) == std::any(b));
+            }
+        }
+
+        BENCHMARK(AnyCompare);
+
+        static void NormalCompare(benchmark::State& state)
+        {
+            srand(time(nullptr));
+
+            for (auto _ : state)
+            {
+                auto a = rand();
+                auto b = rand();
+
+                bool result;
+                benchmark::DoNotOptimize(result = a == b);
+            }
+        }
+
+        BENCHMARK(NormalCompare);
+    }
+
+    namespace Range
+    {
+        namespace Internal
```

Comment:
The `Internal` namespace is needed only in header files, anonymous namespaces can be used in cpp files instead - they give all symbols inside them internal linking, i.e. they will not get outside.
```cpp
namespace {
  inline constexpr int test_size = 10000;
}
void foo() {
  print(test_size);
}
```

Reply:
This is header-only template library. So we cannot use anonymous namespaces?